.github/workflows/build_and_push_tags.yml

```diff
@@ -0,0 +1,38 @@
+name: Renku Docker Image CI - MASTER
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '**'
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        RENKU_VERSION: ["", "0.8.2", "0.9.0"]
+        EXTENSIONS: ["base", "r", "r-bioconductor_3_10", "cuda-9.2", "cuda-10.0-tensorflow-1.14"]
+
+    steps:
+
+    - name: Docker Login
+      uses: Azure/docker-login@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    - uses: actions/checkout@v2
+    - name: Build the Renku Docker pull request images for testing
+      run: |
+        export RENKU_TAG_NAME=$(echo $GITHUB_REF | cut -d / -f 3)
```

Comment:
If the only difference is the tag name why not have this as a conditional step at the top and export it as an environment variable? Then you only need one file/workflow and can reuse almost all of the code.